### PR TITLE
xds/rbac: enforce strict presence-based short-circuit in authenticatedMatcher

### PR DESCRIPTION
**Pull request by al4an444**
This PR fixes a bug in the xDS RBAC authenticatedMatcher where it incorrectly falls through URI/DNS SANs to the Subject DN. According to gRFC A41 and Envoy's specification, only the first non-empty identity source must be consulted. This bug allows an authorization bypass in certain scenarios. A regression test is included.

RELEASE NOTES:
- xds/rbac: Fix `Authenticated` matcher to use the first non-empty identity source (URI SAN, DNS SAN, and Subject DN) in the order specified in gRFC A41.

### CODE REVIEW

File: internal/xds/rbac/rbac_engine_test.go
Comment (easwars) [inline internal/xds/rbac/rbac_engine_test.go]: While you are here, could you please fix this comment about the test. Thanks.

Comment (easwars) [inline internal/xds/rbac/rbac_engine_test.go]: Same here. Could you please fix this comment as well. Thanks.

File: internal/xds/rbac/rbac_engine_test.go [hunk 1]
```diff
@@ -1932,3 +1932,101 @@ func createXDSTypedStruct(t *testing.T, in map[string]any, name string) *anypb.A
...
+		uris            []string
+		dnsNames        []string
+		subjectCN       string
+		matcherContains string
+		wantMatch       bool
+	}{
+		{
+			name:            "uriSANPresentAndMatches",
+			uris:            []string{"spiffe://corp.example/svc/legitimate"},
+			dnsNames:        nil,
```
Comment (easwars) [inline internal/xds/rbac/rbac_engine_test.go]: Nit: Can you please not specify fields in the table that are set to their zero values. This will keep the size of the table down. Thanks.

**Review by easwars — APPROVED** [review]
LGTM, modulo minor nits

**Comment by easwars** [comment]
@arjan-bal for second set of eyes

**Comment by easwars** [comment]
@arjan-bal : I also added a release note. Please review that too. Thanks.

**Comment by al4an444** [comment]
Thanks for the review. I addressed the nits by removing the zero-value fields from the table cases, cleaning up the nearby test comments, and running gofmt. The targeted RBAC tests pass locally.

File: internal/xds/rbac/rbac_engine_test.go [hunk 2]
```diff
@@ -1111,8 +1111,8 @@ func (s) TestChainEngine(t *testing.T) {
 				},
 			},
 		},
-		// This test tests that when there are no SANs or Subject's
-		// distinguished name in incoming RPC's, that authenticated matchers
+		// This test verifies that when there are no SANs or Subject
+		// distinguished name in incoming RPCs, authenticated matchers
 		// match against the empty string.
 		{
 			name: "default-matching-no-credentials",
```
Comment (arjan-bal) [inline internal/xds/rbac/rbac_engine_test.go:1118]: The intention of this test was to ensure that an empty string ("") matches when there are no peer certificates. The fix now correctly returns `PermissionDenied`. Could you add another test case where the `PeerCertificates` list is empty? This should result in `codes.OK`.

File: internal/xds/rbac/rbac_engine_test.go [hunk 3]
```diff
@@ -1932,3 +1931,92 @@ func createXDSTypedStruct(t *testing.T, in map[string]any, name string) *anypb.A
...
+			matcherContains: "svc.legitimate.internal",
+			wantMatch:       true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var uriSANs []*url.URL
+			for _, u := range tt.uris {
+				parsed, _ := url.Parse(u)
```
Comment (arjan-bal) [inline internal/xds/rbac/rbac_engine_test.go]: Let's not silently ignore failures here. We should call `t.Fatalf` to surface issues with the test parameters.

**Review by arjan-bal — COMMENTED** [review]
Mostly looks good, some minor comments related to the tests.

**Comment by al4an444** [comment]
All good, I added a case covering empty PeerCertificates, which expects OK, and updated the URL parsing in the identity precedence test to fail with t.Fatalf on parse errors. The RBAC package tests pass locally.

**Review by arjan-bal — APPROVED** [review]
LGTM!
